src/chapterCleaner.ts

```diff
@@ -0,0 +1,148 @@
+import createDOMPurify from 'dompurify';
+import urlJoin from 'url-join';
+
+import { Manifest } from './types';
+
+export type ExecuteParams = {
+  rawChapter: string;
+  manifest: Manifest;
+  contentPath: string;
+  imageRoot?: string;
+  linkRoot?: string;
+};
+
+export class ChapterCleaner {
+  private static mockHost = 'https://mock.host';
+
+  public static execute(params: ExecuteParams): string {
+    const DOMPurify = createDOMPurify(window);
+
+    const {
+      rawChapter,
+      manifest,
+      contentPath,
+      linkRoot = '',
+      imageRoot = '',
+    } = params;
+
+    DOMPurify.addHook('uponSanitizeElement', (currentNode, data) => {
```

Comment:
I tried to repeat previous logic but it is still not clear for me, why we need this handling for src and href attributes.

Reply:
Could you explain to me why we need to reform href and src in that way?